app-main/public/js/init/setUser.test.js

```diff
@@ -0,0 +1,15 @@
+import setUser from "./setUser.js";
+
+test("Tests setting user data when local storage is empty", () => {
+	document.body.innerHTML = `
+    <div class="avatar">
+      <img referrerpolicy="no-referrer" alt="avatar" />
+    </div>
+  `;
+
+	const avatarElement = document.querySelector(".avatar img");
+
+	setUser();
+
+	expect(avatarElement.attributes.src).not.toBe(undefined);
```

Comment:
This is a good test! I think we could improve this by testing whether user data is persisted to local storage or not. 

I can add a ticket for this so no need to address now. I think we would need to create a local storage mock – something like this:

```
class LocalStorageMock {
  constructor() {
    this.store = {};
  }
  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}
```

Reply:
This is a great idea I had not considered.  If you make a ticket and no one hops on it relatively soon I'll reach out about adding that functionality.

Reply:
Sounds good – I'll create a ticket once this is merged!